fix(docs): Migrate deprecated new Buffer() constructors to new APIs (#116)
nodejs/website-redesign: documentation/0051-node-buffers/index.md

## How to create a buffer

A buffer is created using the [`Buffer.from()`](https://nodejs.org/api/buffer.html#buffer_buffer_from_buffer_alloc_and_buffer_allocunsafe), [`Buffer.alloc()`](https://nodejs.org/api/buffer.html#buffer_class_method_buffer_alloc_size_fill_encoding), and [`Buffer.allocUnsafe()`](https://nodejs.org/api/buffer.html#buffer_class_method_buffer_allocunsafe_size) methods.

```js
const buf = Buffer.from('Hey!')
```
- [`Buffer.from(array)`](https://nodejs.org/api/buffer.html#buffer_class_method_buffer_from_array)
- [`Buffer.from(arrayBuffer[, byteOffset[, length]])`](https://nodejs.org/api/buffer.html#buffer_class_method_buffer_from_arraybuffer_byteoffset_length) 
- [`Buffer.from(buffer)`](https://nodejs.org/api/buffer.html#buffer_class_method_buffer_from_buffer)  

> - [`Buffer.from(string[, encoding])`](https://nodejs.org/api/buffer.html#buffer_class_method_buffer_from_string_encoding)

Reviewer: nit: L32 - L35 can be gotten rid of. Reasons:
1. Seeing these intricate details can look intimidating to a beginner.
2. Advanced users can always look up this info up in the API docs.

- [`Buffer.from(buffer)`](https://nodejs.org/api/buffer.html#buffer_class_method_buffer_from_buffer)  
- [`Buffer.from(string[, encoding])`](https://nodejs.org/api/buffer.html#buffer_class_method_buffer_from_string_encoding)

> You can optionally pass the encoding in the second parameter (defaults to UTF-8).

Reviewer: nit: This line can be retained. It can be useful to know, especially given that now we'll have an example for only this one form of using `Buffer.from()`.

> You can also just initialize the buffer passing the size. This creates a 1KB buffer:

Reviewer: nit: Minor rewording to better fit the narrative.

```suggestion
You can also initialize a buffer with a size. This creates a 1KB buffer:
```

const buf = Buffer.alloc(1024)
//or

> const buf = Buffer.allocUnsafe(1024)

Reviewer: nit: Can be gotten rid of. The reason is that we probably don't want to be teaching `Buffer.allocUnsafe()` to beginners in introductory material. 

Because this is intended to be a getting started guide, it's okay (I hope!) to omit information that advanced users can look up in the API docs.
